cradle-cassandra/src/main/java/com/exactpro/cradle/cassandra/iterators/DuplicateSkippingConvertingPagedIterator.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2023 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exactpro.cradle.cassandra.iterators;
+
+import com.datastax.oss.driver.api.core.MappedAsyncPagingIterable;
+import com.datastax.oss.driver.api.core.cql.Row;
+import com.exactpro.cradle.cassandra.retries.SelectQueryExecutor;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+
+public class DuplicateSkippingConvertingPagedIterator<R, E> extends ConvertingPagedIterator<R, E>{
```

Comment:
Maybe move this logic into the DuplicateSkippingIterator class instead of create new class. I don't know case where we can calculate unique hash.